src/node/utils/Settings.js

```diff
@@ -298,6 +298,24 @@ exports.scrollWhenFocusLineIsOutOfViewport = {
  */
 exports.exposeVersion = false;
 
+
+/*
+ * Import Rate limiting, File size and If a session is required to import
+ *
+ * importRateLimiting defaults at 10 requests per IP per 90 seconds.  After this it's rejected, value is in mS.  See express-rate-limit for more options
```

Comment:
Question: did you test this feature behind a reverse proxy?

Is the middleware smart enough to honor the [`X-Forwarded-For`](https://en.wikipedia.org/wiki/X-Forwarded-For) Header? Otherwise the source IP will always be the reverse proxy's.

Reply:
It can be included in the json blob no?  

https://github.com/ded/express-limiter/blob/master/README.md#examples

Reply:
Don't we already have a setting that has to be set when running behind a reverse proxy (or am I wrong)?

- if not, let's add the configuration in `config.json.template`.
- if yes, let's verify if the middleware confarms to it (it should be an express-level configuration). If it does not, I say it's on us to further configure the rate limiter, without asking anything to the users.

Reply:
Found the setting. Its name is `trustProxy`:

https://github.com/ether/etherpad-lite/blob/83d72d27a405f63782803e0168b1e9121f3cd650/settings.json.template#L302-L311

Reply:
> It can be included in the json blob no?
> https://github.com/ded/express-limiter/blob/master/README.md#examples
>
> this is why I think we should keep the original ms approach instead of seconds :)

Ugh, that's another middleware :)

The rate limiter you used is this one: https://github.com/nfriedly/express-rate-limit

Reply:
Lol yeah sorry :|  It's been a while!

Reply:
https://github.com/nfriedly/express-rate-limit/issues/165

Reply:
tldr; we'd need to do ``app.set('trust proxy', 1);``, that would need to be a separate setting in itself?

Reply:
> tldr; we'd need to do app.set('trust proxy', 1);, that would need to be a separate setting in itself?

I think the issue you mentioned (nfriedly/express-rate-limit#165) solves the question. `express-rate-limit` will always look at the `ip` field but - when told so via `trust proxy` setting, Express will automatically fill that value with `x-forwarded-for`.

Our `trustProxy` settings already does exactly that. I'll document it in the source.

Resolving the conversation.